### PR TITLE
[FEATURE] Add plugin to allow oidc authentication without login box

### PR DESCRIPTION
**Pull request by SomeBdyElse**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,25 @@ If you are a Swiss Alpine Club section, be sure to get in touch with Bern in ord
 secret.
 
 
+## Default FE Loginbox
+
+This extension integrations with the felogin systemextension and provides a new marker ###OPENID_CONNECT### to be used
```
Comment (xperseguers) [inline README.md]: - integrations => integrates
- systemextension => system extension

File: README.md [hunk 2]
```diff
@@ -8,6 +8,25 @@ If you are a Swiss Alpine Club section, be sure to get in touch with Bern in ord
 secret.
 
 
+## Default FE Loginbox
+
+This extension integrations with the felogin systemextension and provides a new marker ###OPENID_CONNECT### to be used 
+in the felogin template. A sample template is included. Once the users clicks the new openid connect link, he/she will
```
Comment (xperseguers) [inline README.md]: - the users clicks => the user clicks
- he/she => she (I figured out by reading many presentations/texts from native speakers that it's very common to use the feminine form when speaking about a user)

File: README.md [hunk 3]
```diff
@@ -8,6 +8,25 @@ If you are a Swiss Alpine Club section, be sure to get in touch with Bern in ord
 secret.
 
 
+## Default FE Loginbox
+
+This extension integrations with the felogin systemextension and provides a new marker ###OPENID_CONNECT### to be used 
+in the felogin template. A sample template is included. Once the users clicks the new openid connect link, he/she will 
+redirected to the ids server.
```
Comment (xperseguers) [inline README.md]: - ids => IDS

File: README.md [hunk 4]
```diff
@@ -8,6 +8,25 @@ If you are a Swiss Alpine Club section, be sure to get in touch with Bern in ord
...
 
+## Default FE Loginbox
+
+This extension integrations with the felogin systemextension and provides a new marker ###OPENID_CONNECT### to be used 
+in the felogin template. A sample template is included. Once the users clicks the new openid connect link, he/she will 
+redirected to the ids server. 
+
+## OIDC Login
+
+If openid_connect is your only means of fe login, you can use the included "OIDC Login" plugin. Add it to your login
```
Comment (xperseguers) [inline README.md]: - fe login => either FE login or better Frontend login

File: Classes/Controller/LoginController.php [hunk 5]
```diff
@@ -0,0 +1,108 @@
+<?php
+
```
Comment (xperseguers) [inline Classes/Controller/LoginController.php]: Missing standard TYPO3 header block

File: Classes/Controller/LoginController.php [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+     * TypoScript configuratoin of this plugin
+     *
+     * @var array
+     */
+    protected $pluginConfiguration;
+
+    /**
+     * @var ContentObjectRenderer
+     *
+     * Will be injected automatically, if this controller is called as a plugin
```
Comment (xperseguers) [inline Classes/Controller/LoginController.php]: Comment should go before the `@var` declaration

File: Classes/Controller/LoginController.php [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    public function __construct()
+    {
+        $this->settings = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['oidc']);
+    }
+
+    /**
+     * @param string $_ ignored
+     * @param array|null $pluginConfiguration
+     *
+     * Main entry point for the OIDC plugin.
```
Comment (xperseguers) [inline Classes/Controller/LoginController.php]: comment goes *before* the parameters

File: Classes/Controller/LoginController.php [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+     *
+     * If the user is not logged in, redirect to the authorization server to start the oidc process
+     *
+     * If the user has just been logged in and just came back from the authorization server, redirect the user to the
+     * final redirect URL.
+     */
+    public function login($_ = '', ?array $pluginConfiguration)
```
Comment (xperseguers) [inline Classes/Controller/LoginController.php]: `?array` is not compatible with PHP 5.5 which is still a dependency defined in `composer.json`

Note that I'd be fine to raise the dependency to PHP 7.2, and drop TYPO3 7.6 support as well but this needs to be done separately.

File: Classes/Controller/LoginController.php [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        if (!empty($pluginConfiguration)) {
+            $this->pluginConfiguration = $pluginConfiguration;
+        }
+
+        if (GeneralUtility::_GP('logintype') == 'login') {
+            $this->performRedirectAfterLogin();
+            // performRedirectAfterLogin stops flow by emitting a redirect
```
Comment (xperseguers) [inline Classes/Controller/LoginController.php]: I'd prefer the comment before the call

File: Classes/Controller/LoginController.php [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+        $authorizationUrl = $service->getAuthorizationUrl();
+
+        if (session_id() === '') {
+            session_start();
+        }
+
+        $state = $service->getState();
+        $_SESSION['oidc_state'] = $state;
+        $_SESSION['oidc_login_url'] = $_SERVER['REQUEST_URI'];
```
Comment (xperseguers) [inline Classes/Controller/LoginController.php]: Any reason for not using the official TYPO3 API `GeneralUtility::getIndpEnv()`?

**Comment by SomeBdyElse** [comment]
@xperseguers Thank you for the valuable comments. I addressed them by: 
* Updating the README, correcting my spelling
* Fixed comments in the new controller class
* Changed typing to remain compatible with the older PHP version
* Removed an unused import (unrelated)

**Comment by SomeBdyElse** [comment]
My last commit is a "fixup". I suggest to **squash**-merge the feature branch into master.

**Pull request merged**